deploy/lib/server_config.rb

```diff
@@ -2451,7 +2451,8 @@ def xcc
           :http_connection_retry_count => @properties["ml.http.retry-count"].to_i,
           :http_connection_open_timeout => @properties["ml.http.open-timeout"].to_i,
           :http_connection_read_timeout => @properties["ml.http.read-timeout"].to_i,
-          :http_connection_retry_delay => @properties["ml.http.retry-delay"].to_i
+          :http_connection_retry_delay => @properties["ml.http.retry-delay"].to_i,
+          :use_https_for_rest => @properties["ml.ssl-certificate-template"].present? || @properties["ml.use-https-for-rest"] == "true"
```

Comment:
The rest server that gets called by Roxy is typically the server at app-port, to which ssl-cert stuff is typically applied. The xcc server could be a separate server, and it is more difficult to guess if it is running https or not. Easiest way to solve that would be to introduce a `use-https-for-xcc` property in default.properties, which can be overridden in the other ones. The `use_https_for_rest` (or better: `use_https_for_xcc`) would only respond to that property in the case of XCC, unless perhaps the `install-xcc` property has been set to false, and xcc-port equals app-port (a helper function to verify this would be convenient).

Reply:
I wasn't sure whether to use the `use-https-for-rest` or just the `use-https` option for xcc, so I went with what #862 mentioned. I agree with you though, it's best to introduce a separate option for xcc in particular. I'll amend the PR.